pandas/io/common.py

```diff
@@ -150,6 +150,23 @@ def urlopen(*args, **kwargs):
     return urllib.request.urlopen(*args, **kwargs)
 
 
+def get_fs_for_path(filepath):
```

Comment:
can you type this (and the return annotation)

Reply:
Left return type for now since it include optional dependencies.

e.g ```Union[s3fs.S3FileSystem, gcsfs.GCSFileSystem, None]```

Can add imports to the TYPE_CHECKING block at the top if that's appropriate?